### DISCUSSION TITLE
Widget + App transparency

### DISCUSSION THREAD

**Discussion by MiX-4401** [Textualize/textual#5288] [Q&A]
I would like my textual app to be see-through to whatever is behind it. I have all the transparency settings enabled for my compositor and terminal emu. And generally have very clean, slight transparency on my desktop + for all apps. 

Textual puts a large background colour over the 'screen' of the App() and I would much prefer transparency options. Is it possible to remove the styles.background of widgets such that they would become transparent. Also, what about screen.styles.background?

**Comment by willmcgugan** [comment]
You can't remove the background, as all text has a background. You can set the background to `ansi_default` if you enable ANSI colors on the app. This is will probably give you the effect you are looking for. 

However, I would recommend against it. ANSI themes make it [impossible](https://textual.textualize.io/FAQ/#why-doesnt-textual-support-ansi-themes) to create apps that look good for everyone.


> Also, what about screen.styles.background?

What about it?

**Reply by MiX-4401** [reply]
Yeah cool. I'll try ANSI colours and see how it goes. So screen.styles.background; I see that in the [docs](https://textual.textualize.io/guide/app/) there is an example showing a variety of colours to set screen.styles.background to. I expect this is the bottom-most background of the App. And thus a background I would like to make transparent. This is what I am referring to. 

Thanks you getting back to me.

**Reply by MiX-4401** [reply]
![example](https://github.com/user-attachments/assets/5a00273d-d1bd-4ee2-b876-685506cbb804)

I also see that there is almost like a padding around the edges of the App. Is it possible to remove this padding such that the Textual app spans across in a snug fit within my terminal emu.?

I do not believe that this is my terminal emu causing this padding either.

**Reply by TomJGooding** [reply]
> I also see that there is almost like a padding around the edges of the App. Is it possible to remove this padding such that the Textual app spans across in a snug fit within my terminal emu.?

Sorry but I'm not seeing any padding in your screenshot  -  can you clarify what you mean?

**Reply by MiX-4401** [reply]
I mean the padding present between the very top of the scrnshot to the start of the App's background. This padding is present on all edges of the textual app. I have my Textual App in fullscreen in the scrnshot to make it clearer. You can see that my compositor is trying to make the app transparent, hence why you see a bit of my background (the white splotch on the bottom right corner).

![example2](https://github.com/user-attachments/assets/73a7716c-b3f7-4156-b80b-5eaf5222cf8b)


You can see, when my compositor applies a border around the boundaries of the textual app, with the assistance of all the blue, there is a clear padding between the true-border, and the background of the app. Between this space there is absolute transparency.

This is the crux of why I am looking for transparency options, my compositor is fighting against the textual app whereby textual has a solid background, but my compositor is trying to make it transparent. So I have this background colour against a transparent border. As the app is in fullscreen it is very clear were there is transparency between the *padded???? border and the app. 

I am not to sure how to eliminate this padded border. :).

**Reply by TomJGooding** [reply]
Thanks for clarifying. That looks to me like the border in your terminal emulator - have you checked the settings?
